lib/YaST/EFItools.pm

```diff
@@ -0,0 +1,41 @@
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package YaST::EFItools;
+
+use strict;
+use warnings;
+use Exporter 'import';
+use testapi;
+use scheduler 'get_test_suite_data';
+use Test::Assert ':all';
+
+our @EXPORT = qw(
+  check_secure_boot_status
+);
+
+use constant MIN_OCTALS => 5;
+
+=head1 tools for EFI related stuff
+
+=head2 check_secure_boot_status
+
+The function reads out the octal values of the variable in efivars/SecureBoot-*.
+Secure Boot status is stored in the 5th byte of that variable (0=disabled, 1=enabled)
+See https://www.codelinsoft.it/sito/index.php/support/wiki/secure-boot for details
+
+We check if the 5th byte is set to 0 or 1 and act accordingly. 
+
+=cut
+
+sub check_secure_boot_status {
+    my $test_data = get_test_suite_data();
+    assert_script_run("ls /sys/firmware/efi/efivars/SecureBoot-*");
+    my $octal_str = script_output("od -An -t u1 /sys/firmware/efi/efivars/SecureBoot-*");
+    my @octal_array = split(/\s+/, $octal_str);
+    die 'Unexpected values in Secure Boot file' unless 0 + @octal_array >= MIN_OCTALS && $octal_array[MIN_OCTALS - 1] =~ /0|1/;
+    my $secure_boot = $octal_array[MIN_OCTALS - 1] ? 'enabled' : 'disabled';
+    assert_equals($test_data->{secure_boot}, $secure_boot, "The secure boot option is not $test_data->{secure_boot}");
```

Comment:
don't forget the test design, test data is read in Test Layer along with assertion above, you just need to library to calculate something, it would be something like "read_secure_boot_status" where you document the potential result, but never "check".

Reply:
Changed